src/tests/ngxs.setup.spec.ts

```diff
@@ -136,3 +142,81 @@ describe('Full testing NGXS States with NgxsTestBed', () => {
         expect(getStateContextMocks[FOOD_STATE_NAME].dispatch).toHaveBeenCalledTimes(1);
     });
 });
+
+describe('Select tests', () => {
+    class FeedAction {
+        public static type = 'zoo';
+        constructor(public payload: number) {}
+    }
+
+    class AddAnimalsAction {
+        public static type = 'add animals';
+        constructor(public payload: number) {}
+    }
+
+    @State({ name: 'zoo', defaults: { feed: 0, animals: 0 } })
+    class ZooState {
+        @Selector()
+        static feed(state: { feed: number; animals: number }) {
+            return state.feed;
+        }
+
+        @Selector()
+        static animals(state: { feed: number; animals: number }) {
+            return state.animals;
+        }
+
+        @Action(FeedAction) public feed(ctx: StateContext<any>, { payload }: FeedAction) {
+            ctx.patchState({ feed: payload });
+        }
+
+        @Action(AddAnimalsAction) public animals(ctx: StateContext<any>, { payload }: AddAnimalsAction) {
+            const state = ctx.getState();
+            ctx.patchState({ animals: state.animals + payload });
+        }
+    }
+
+    @Component({
+        template: `
+            <span>{{ foodSelector$ | async }}</span>
+            <p>{{ animalsSelector$ | async }}</p>
+        `
+    })
+    class HostComponent {
+        @Select(ZooState.feed) foodSelector$!: Observable<number>;
+        @Select(ZooState.animals) animalsSelector$!: Observable<number>;
+    }
+
+    let foodSelectorSubject: Subject<number>;
+    let animalsSelectorSubject: Subject<number>;
+    let fixture: ComponentFixture<HostComponent>;
+    let component: HostComponent;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [HostComponent],
+            imports: [CommonModule, NgxsModule.forRoot()]
```

Comment:
why `forRoot()` instead `forRoot([ZooState])`

Reply:
We don't need to import it as we are mocking the selectors. 
By doing so, we don't have to import every dependency of our states. 
With this you can test you component by testing wheter or not you are dispatching the correct actions or displaying the correct template value when an event occurs from a selector, an user input or something like that. 
It's my approach to unit testing, but it should still work for tests including the side effects triggered by the store, both can work together.

Reply:
it’s just confusing if I test the condition itself